vispy/visuals/mesh.py

```diff
@@ -166,42 +88,20 @@ class MeshVisual(Visual):
         The values to use for each vertex (for colormapping).
     meshdata : instance of MeshData | None
         The meshdata.
-    shading : str | None
-        Shading to use.
     mode : str
         The drawing mode.
     **kwargs : dict
         Keyword arguments to pass to `Visual`.
     """
     def __init__(self, vertices=None, faces=None, vertex_colors=None,
-                 face_colors=None, color=(0.5, 0.5, 1, 1), vertex_values=None,
-                 meshdata=None, shading=None, mode='triangles', **kwargs):
-
-        # Function for computing phong shading
-        # self._phong = Function(phong_template)
-
-        # Visual.__init__ -> prepare_transforms() -> uses shading
-        self.shading = shading
-
-        if shading is not None:
-            Visual.__init__(self, vcode=shading_vertex_template,
-                            fcode=shading_fragment_template,
-                            **kwargs)
-
-        else:
-            Visual.__init__(self, vcode=vertex_template,
-                            fcode=fragment_template,
-                            **kwargs)
-
-        self.set_gl_state('translucent', depth_test=True,
-                          cull_face=False)
+                 face_colors=None, color=(1, 1, 1, 1), vertex_values=None,
+                 meshdata=None, mode='triangles', **kwargs):
```

Comment:
I didn't see this outline anywhere in the PR, but why did the default color change? Pure white seems risky to work for all examples that we have. And `shading` was removed, are users expected to add the filter themselves? Would it be possible to add the filter here if `shading` is specified? Or are there some other side effects to this?

Reply:
Indeed, good point. The removed `shading` param breaks backward compatibility and is maybe not what we want.
I initially assumed users would add the filters manually so I removed these options from the MeshVisual.
I don't see any reason why it wouldn't work to add the shading filter here depending on `shading`.

Reply:
We could keep the shading keyword in the MeshVisual but print a deprecation warning and point to the example to show how to attach the ShadingFilter to a created MeshVisual. Then in the future we could remove it.

Either way it is probably a good idea if we add to the MeshVisual docstring the various filters that apply to the MeshVisual. Thoughts? @kmuehlbauer @larsoner

Reply:
Indeed the behavior should be documented. I'm fine with leaving the option in the `mesh` constructor, both for backward compat and going forward. It's not too hard to leave it in there (it should only be a one or two-liner, right?) so little maintenance cost. And in the docs for Mesh you can always tell people that for advanced/other options, they should use a filter instead.

Reply:
@djhoese I would second @larsoner.

Reply:
@djhoese I'd also go for @larsoner's proposal. Keeping the `shading` keyword in the constructor for backward compatibility and ease of use, while allowing customisation with `shading=None`.